# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: testing/web-platform/meta/service-workers/idlharness.https.any.js.ini
# Commit: 244df402d6eb
# Full Hash: 244df402d6ebc15d6845c4144245511021408835
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1131324
# File Overlap Count: 1
# Description:
#   Bug 1131324 - Expose ServiceWorkerContainer on WorkerNavigator. r=dom-worker-reviewers,webidl,smaug
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213725
# ==============================================================================

diff -r 57f70ba5f27e -r 244df402d6eb testing/web-platform/meta/service-workers/idlharness.https.any.js.ini
--- a/testing/web-platform/meta/service-workers/idlharness.https.any.js.ini	Thu Oct 24 03:02:39 2024 +0000
+++ b/testing/web-platform/meta/service-workers/idlharness.https.any.js.ini	Thu Oct 24 03:02:40 2024 +0000
@@ -1,233 +1,23 @@
 [idlharness.https.any.sharedworker.html]
-  [ServiceWorkerContainer interface: operation startMessages()]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "startMessages()" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "controller" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface object length]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "oncontrollerchange" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessageerror" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistrations()]
-    expected: FAIL
-
-  [Stringification of navigator.serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute onmessage]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface object name]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's @@unscopables property]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: calling getRegistration(optional USVString) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute ready]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistration(optional USVString)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistration(optional USVString)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute onmessageerror]
-    expected: FAIL
-
-  [WorkerNavigator interface: attribute serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's "constructor" property]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute oncontrollerchange]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute controller]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessage" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer must be primary interface of navigator.serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "ready" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistrations()" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation register((TrustedScriptURL or USVString), optional RegistrationOptions)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "register((TrustedScriptURL or USVString), optional RegistrationOptions)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: calling register((TrustedScriptURL or USVString), optional RegistrationOptions) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
 
 [idlharness.https.any.worker.html]
-  [ServiceWorkerContainer interface: operation startMessages()]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "startMessages()" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "controller" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface object length]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "oncontrollerchange" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessageerror" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistrations()]
-    expected: FAIL
-
-  [Stringification of navigator.serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute onmessage]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface object name]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's @@unscopables property]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: calling getRegistration(optional USVString) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute ready]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistration(optional USVString)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistration(optional USVString)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute onmessageerror]
-    expected: FAIL
-
-  [WorkerNavigator interface: attribute serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's "constructor" property]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute oncontrollerchange]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute controller]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessage" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer must be primary interface of navigator.serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "ready" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistrations()" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation register((TrustedScriptURL or USVString), optional RegistrationOptions)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "register((TrustedScriptURL or USVString), optional RegistrationOptions)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: calling register((TrustedScriptURL or USVString), optional RegistrationOptions) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
 
 [idlharness.https.any.serviceworker.html]
   [ServiceWorker interface: serviceWorker must inherit property "scriptURL" with the proper type]
     expected: FAIL
 
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "startMessages()" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "controller" with the proper type]
-    expected: FAIL
-
   [ServiceWorker interface: calling postMessage(any, sequence<object>) on serviceWorker with too few arguments must throw TypeError]
     expected: FAIL
 
-  [ServiceWorkerContainer interface object length]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "oncontrollerchange" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation startMessages()]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessageerror" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistrations()]
-    expected: FAIL
-
   [Stringification of serviceWorker]
     expected: FAIL
 
-  [Stringification of navigator.serviceWorker]
-    expected: FAIL
-
   [FetchEvent interface: attribute replacesClientId]
     expected: FAIL
 
   [ServiceWorkerGlobalScope interface: self must inherit property "serviceWorker" with the proper type]
     expected: FAIL
 
-  [ServiceWorkerContainer interface: attribute onmessage]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface object name]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's @@unscopables property]
-    expected: FAIL
-
   [ServiceWorker interface: serviceWorker must inherit property "onstatechange" with the proper type]
     expected: FAIL
 
@@ -240,54 +30,12 @@
   [ServiceWorker interface: serviceWorker must inherit property "postMessage(any, sequence<object>)" with the proper type]
     expected: FAIL
 
-  [ServiceWorkerContainer interface: calling getRegistration(optional USVString) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute ready]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistration(optional USVString)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: operation getRegistration(optional USVString)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute onmessageerror]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object's "constructor" property]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute oncontrollerchange]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: attribute controller]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "onmessage" with the proper type]
-    expected: FAIL
-
-  [WorkerNavigator interface: attribute serviceWorker]
-    expected: FAIL
-
   [ServiceWorkerGlobalScope interface: attribute serviceWorker]
     expected: FAIL
 
   [WindowClient interface: attribute ancestorOrigins]
     expected: FAIL
 
-  [ServiceWorkerContainer must be primary interface of navigator.serviceWorker]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "ready" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: existence and properties of interface prototype object]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "getRegistrations()" with the proper type]
-    expected: FAIL
-
   [FetchEvent interface: new FetchEvent("type", { request: new Request("") }) must inherit property "replacesClientId" with the proper type]
     expected: FAIL
 
@@ -318,14 +66,5 @@
   [InstallEvent interface: operation addRoutes((RouterRule or sequence<RouterRule>))]
     expected: FAIL
 
-  [ServiceWorkerContainer interface: operation register((TrustedScriptURL or USVString), optional RegistrationOptions)]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: navigator.serviceWorker must inherit property "register((TrustedScriptURL or USVString), optional RegistrationOptions)" with the proper type]
-    expected: FAIL
-
-  [ServiceWorkerContainer interface: calling register((TrustedScriptURL or USVString), optional RegistrationOptions) on navigator.serviceWorker with too few arguments must throw TypeError]
-    expected: FAIL
-
 
 [idlharness.https.any.html]